src/HttpClientFactory/Polly/src/DependencyInjection/PollyHttpClientBuilderExtensions.cs

```diff
@@ -234,5 +234,55 @@ public static IHttpClientBuilder AddPolicyHandlerFromRegistry(this IHttpClientBu
             builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(policy));
             return builder;
         }
+
+        /// <summary>
+        /// Adds a <see cref="PolicyHttpMessageHandler"/> which will surround request execution with a policy returned
+        /// by executing provided key selection logic <paramref name="policySelector"/>. Based on the 
+        /// </summary>
+        /// <param name="builder">The <see cref="IHttpClientBuilder"/>.</param>
+        /// <param name="keySelector">To select a Policy key based on <seealso cref="HttpRequestMessage"/>.</param>
+        /// <param name="policySelector">Selects an <see cref="IAsyncPolicy{HttpResponseMessage}"/> to apply to the current request based on key selection.</param>
```

Comment:
This should be called `policyFactory` since it's used to instantiate and store the policy if it's not present.